server/protocols/webadmin/webadmin-mailqueue/src/main/java/org/apache/james/webadmin/service/DeleteMailsFromMailQueueTask.java

```diff
@@ -95,49 +98,72 @@ public UnknownSerializedQueue(String queueName) {
         }
     }
 
+    @FunctionalInterface
+    public interface MailQueueFactory {
+        ManageableMailQueue create(MailQueueName mailQueueName) throws MailQueue.MailQueueException;
+    }
+
     public static final TaskType TYPE = TaskType.of("delete-mails-from-mail-queue");
 
-    private final ManageableMailQueue queue;
     private final Optional<MailAddress> maybeSender;
     private final Optional<String> maybeName;
     private final Optional<MailAddress> maybeRecipient;
-
-    private final long initialCount;
-
-    public DeleteMailsFromMailQueueTask(ManageableMailQueue queue, Optional<MailAddress> maybeSender,
+    private final MailQueueFactory factory;
+    private final MailQueueName queueName;
+    private Optional<Long> initialCount;
+    private Optional<ManageableMailQueue> queue;
+    private Optional<TaskExecutionDetails.AdditionalInformation> lastAdditionalInformation;
+
+    public DeleteMailsFromMailQueueTask(MailQueueName queueName, MailQueueFactory factory,
+                                        Optional<MailAddress> maybeSender,
                                         Optional<String> maybeName, Optional<MailAddress> maybeRecipient) {
```

Comment:
s/maybe/optional/g or s/maybe//g

Reply:
I keep it like that in this commit for consistency and will add another commit to rename all fields and arguments because you are right about the semantic